pkg/grpc/remote_grpc_request_authenticator_test.go

```diff
@@ -0,0 +1,76 @@
+package grpc_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/buildbarn/bb-storage/internal/mock"
+	"github.com/buildbarn/bb-storage/pkg/auth"
+	bb_grpc "github.com/buildbarn/bb-storage/pkg/grpc"
+	auth_pb "github.com/buildbarn/bb-storage/pkg/proto/auth"
+	"github.com/buildbarn/bb-storage/pkg/testutil"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/mock/gomock"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/metadata"
+	"google.golang.org/grpc/status"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func TestRemoteGrpcRequestAuthenticator(t *testing.T) {
+	ctrl, ctx := gomock.WithContext(context.Background(), t)
+	md := metadata.New(
+		map[string]string{
+			"Authorization": "token",
```

Comment:
Doesn't gRPC metadata use lowercase header names?

Reply:
The Go gRPC library is case insensitive and translates them to lower case.

Reply:
Ah, gotcha. It looks like `metadata.New()` does the normalization to lowercase for you, so it's fine.